Compile issues

Actually, from last yaml config patch, it won't compile, and isn't only me.

-
Loading NachoSpigot config from nacho.yml
Loading NachoSpigot config from nacho.yml
Loading NachoSpigot config from nacho.yml
Loading NachoSpigot config from nacho.yml
[ERROR] Tests run: 4, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 0.297 s <<< FAILURE! - in me.elier.nachospigot.config.ConfigurationTests
[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig  Time elapsed: 0.161 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig(ConfigurationTests.java:33)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.loadConfig  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.loadConfig(ConfigurationTests.java:38)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig  Time elapsed: 0.013 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig(ConfigurationTests.java:26)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig  Time elapsed: 0.017 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig(ConfigurationTests.java:16)

I just did it again, this it's the build failure:


        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig  Time elapsed: 0.008 s  <<< ERROR!
java.lang.NullPointerException
        at org.bukkit.Bukkit.getLogger(Bukkit.java:611)
        at me.elier.nachospigot.config.NachoConfig.migrate(NachoConfig.java:64)
        at me.elier.nachospigot.config.NachoConfig.init(NachoConfig.java:49)
        at me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig(ConfigurationTests.java:16)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)

> Try deleting NachoSpigot-Server/target and then building again

Same;:

Loading NachoSpigot config from nacho.yml
[ERROR] Tests run: 4, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 0.408 s <<< FAILURE! - in me.elier.nachospigot.config.ConfigurationTests
[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig  Time elapsed: 0.27 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig(ConfigurationTests.java:33)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.loadConfig  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.loadConfig(ConfigurationTests.java:38)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig  Time elapsed: 0.008 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig(ConfigurationTests.java:26)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig  Time elapsed: 0.026 s  <<< ERROR!
java.lang.NullPointerException
        at me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig(ConfigurationTests.java:16)

Okay, so i did it and seems like the process it's beeing used already


[ERROR] Tests run: 4, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 0.95 s <<< FAILURE! - in me.elier.nachospigot.config.ConfigurationTests
[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig  Time elapsed: 0.608 s  <<< ERROR!
java.lang.RuntimeException:
java.nio.file.FileSystemException: nacho.json: 
The process does not have access to the file because it is being used by another process.

        at me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig(ConfigurationTests.java:37)
Caused by: java.nio.file.FileSystemException:
nacho.json: 
The process does not have access to the file because it is being used by another process.

        at me.elier.nachospigot.config.ConfigurationTests.migrateEmptyConfig(ConfigurationTests.java:37)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.loadConfig  Time elapsed: 0.013 s  <<< ERROR!
java.lang.RuntimeException:
java.nio.file.FileSystemException: nacho.json: El proceso no tiene acceso al archivo porque está siendo utilizado por otro proceso.

        at me.elier.nachospigot.config.ConfigurationTests.loadConfig(ConfigurationTests.java:43)
Caused by: java.nio.file.FileSystemException:
nacho.json: 
The process does not have access to the file because it is being used by another process.

        at me.elier.nachospigot.config.ConfigurationTests.loadConfig(ConfigurationTests.java:43)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig  Time elapsed: 0.023 s  <<< ERROR!
java.lang.RuntimeException:
java.nio.file.FileSystemException: nacho.json: El proceso no tiene acceso al archivo porque está siendo utilizado por otro proceso.

        at me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig(ConfigurationTests.java:29)
Caused by: java.nio.file.FileSystemException:
nacho.json: El proceso no tiene acceso al archivo porque está siendo utilizado por otro proceso.

        at me.elier.nachospigot.config.ConfigurationTests.migratePartialModifiedConfig(ConfigurationTests.java:29)

[ERROR] me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig  Time elapsed: 0.011 s  <<< ERROR!
java.lang.RuntimeException:
java.nio.file.FileSystemException: nacho.json: 
The process does not have access to the file because it is being used by another process.

        at me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig(ConfigurationTests.java:18)
Caused by: java.nio.file.FileSystemException:
nacho.json:
The process does not have access to the file because it is being used by another process.

        at me.elier.nachospigot.config.ConfigurationTests.migrateFullConfig(ConfigurationTests.java:18)

--- a/NachoSpigot-Server/src/main/java/me/elier/nachospigot/config/NachoConfig.java
+++ b/NachoSpigot-Server/src/main/java/me/elier/nachospigot/config/NachoConfig.java
@@ -51,20 +51,14 @@ public class NachoConfig {
         int configVersion = 1; // Update this every new configuration update
         version = getInt("config-version", configVersion);
         set("config-version", configVersion);
-        readConfig(NachoConfig.class, null);
         c.setHeader(HEADER);
         c.addComment("config-version", "Configuration version, do NOT modify this!");
+        readConfig(NachoConfig.class, null);
     }
 
     private static void migrate(File old_config) {
         OldNachoConfig nachoJson = FileUtils.toObject(old_config, OldNachoConfig.class);
-        try {
-            Files.delete(old_config.toPath());
-        } catch (IOException e) {
-            Bukkit.getLogger().log(Level.SEVERE, "Failed to delete nacho.json during migration to nacho.yml");
-            throw Throwables.propagate(e);
-        }
-        if(nachoJson == null) return;
+        if(nachoJson == null) {old_config.delete(); return;}
         set("settings.save-empty-scoreboard-teams", nachoJson.saveEmptyScoreboardTeams);
         set("settings.commands.enable-version-command", nachoJson.enableVersionCommand);
         set("settings.commands.enable-plugins-command", nachoJson.enablePluginsCommand);
@@ -105,6 +99,7 @@ public class NachoConfig {
         set("world-settings.default.disable-sponge-absorption", nachoJson.disableSpongeAbsorption);
         set("settings.fix-eat-while-running", nachoJson.fixEatWhileRunning);
         set("settings.hide-projectiles-from-hidden-players", nachoJson.hideProjectilesFromHiddenPlayers);
+        old_config.delete();
     }
 
     static void readConfig(Class<?> clazz, Object instance) {
@@ -125,6 +120,7 @@ public class NachoConfig {
 
         try {
             config.save(CONFIG_FILE);
+            //c.saveComments(CONFIG_FILE);
         } catch (IOException ex) {
             Bukkit.getLogger().log(Level.SEVERE, "Could not save " + CONFIG_FILE, ex);
         }

--- a/NachoSpigot-Server/src/main/java/me/elier/nachospigot/config/NachoConfig.java
+++ b/NachoSpigot-Server/src/main/java/me/elier/nachospigot/config/NachoConfig.java
@@ -39,17 +39,6 @@ public class NachoConfig {
         CONFIG_FILE = configFile;
         config = new YamlConfiguration();
         try {
-            + "join us in our Discord.\n"
-            + "\n"
-            + "Discord: https://discord.gg/SBTEbSx\n"
-            + "Github: https://github.com/CobbleSword/NachoSpigot\n";
-    static YamlConfiguration config;
-    static int version;
-
-    public static void init(File configFile) {
-        CONFIG_FILE = configFile;
-        config = new YamlConfiguration();
-        try {
             System.out.println("Loading NachoSpigot config from " + configFile.getName());
             config.load(CONFIG_FILE);
         } catch (IOException ignored) {

--- a/NachoSpigot-Server/src/main/java/me/elier/util/Utils.java
+++ b/NachoSpigot-Server/src/main/java/me/elier/util/Utils.java
@@ -1,44 +1,7 @@
 package me.elier.util;
 
-import org.apache.commons.lang.reflect.FieldUtils;
-
-import java.util.Arrays;
-
 public class Utils {
-    // from String.repeat in Java 11
     public static String repeat(String string, int count) {
-        byte[] value = null;
-        byte coder = 0;
-        try {
-            value = (byte[]) FieldUtils.readDeclaredField(string, "value", true);
-            coder = (byte) FieldUtils.readDeclaredField(string, "coder", true);
-        } catch (IllegalAccessException ignored) {}
-        if (count < 0) {
-            throw new IllegalArgumentException("count is negative: " + count);
-        }
-        if (count == 1) {
-            return string;
-        }
-        final int len = value.length;
-        if (len == 0 || count == 0) {
-            return "";
-        }
-        if (Integer.MAX_VALUE / count < len) {
-            throw new OutOfMemoryError("Required length exceeds implementation limit");
-        }
-        if (len == 1) {
-            final byte[] single = new byte[count];
-            Arrays.fill(single, value[0]);
-            return new String(single, coder);
-        }
-        final int limit = len * count;
-        final byte[] multiple = new byte[limit];
-        System.arraycopy(value, 0, multiple, 0, len);
-        int copied = len;
-        for (; copied < limit - copied; copied <<= 1) {
-            System.arraycopy(multiple, 0, multiple, copied, copied);
-        }
-        System.arraycopy(multiple, 0, multiple, copied, limit - copied);
-        return new String(multiple, coder);
+        return new String(new char[count]).replace("\0", string);
     }
 }

